frame/contracts/tests/caller_contract.wat

```diff
@@ -0,0 +1,275 @@
+(module
+    (import "env" "ext_scratch_size" (func $ext_scratch_size (result i32)))
+    (import "env" "ext_scratch_read" (func $ext_scratch_read (param i32 i32 i32)))
+    (import "env" "ext_balance" (func $ext_balance))
+    (import "env" "ext_call" (func $ext_call (param i32 i32 i64 i32 i32 i32 i32) (result i32)))
+    (import "env" "ext_instantiate" (func $ext_instantiate (param i32 i32 i64 i32 i32 i32 i32) (result i32)))
+    (import "env" "ext_println" (func $ext_println (param i32 i32)))
+    (import "env" "memory" (memory 1 1))
+
+    (func $assert (param i32)
+	(block $ok
+	    (br_if $ok
+		(get_local 0)
+	    )
+	    (unreachable)
+	)
+    )
+
+    (func $current_balance (param $sp i32) (result i64)
+	(call $ext_balance)
+	(call $assert
+	    (i32.eq (call $ext_scratch_size) (i32.const 8))
+	)
+	(call $ext_scratch_read
+	    (i32.sub (get_local $sp) (i32.const 8))
+	    (i32.const 0)
+	    (i32.const 8)
+	)
```

Comment:
Please fix the formatting/indentation issues with most of the `.wat` files.

Reply:
Sorry, I somehow lost the formatting while moving the code.